### PR TITLE
Add place 

### PR DESCRIPTION
**Pull request by RQ16**
Renders the add-place view at the endpoint /:lang/add-place, a form layout with inputs...Styling was added to the form

### CODE REVIEW

File: src/controllers/places.js [hunk 1]
```diff
@@ -67,9 +66,69 @@ placeController.getSpecific = (req, res) => {
       localLang: req.app.locals[req.params.lang],
       lang: req.params.lang,
       dir
-
     });
   });
 };
 
+placeController.renderForm = (req, res) => {
+  if (req.params.lang !== 'en' && req.params.lang !== 'ar') {
```
Comment (m4v15) [inline src/controllers/places.js]: Factor this out?

File: src/controllers/places.js [hunk 2]
```diff
@@ -67,9 +66,69 @@ placeController.getSpecific = (req, res) => {
...
+    return res.status(404).send('Page does not exist');
+  }
+
+  let dir = '';
```
Comment (m4v15) [inline src/controllers/places.js]: And this?

File: src/controllers/places.js (hunk 1 above)
Comment (m4v15) [reply src/controllers/places.js]: Potentially as middleware that you can add to routes:

```js
(req,res,next) => {
    // set req.locals
   next()
}
```

File: src/controllers/getLatLng.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const queryString = require('querystring');
```
Comment (m4v15) [inline src/controllers/getLatLng.js]: Is this a controller? maybe put it in a helpers folder instead?

**Comment by Suuuuha** [comment]
@m4v15 I guess we're done, you can review it now :)

File: public/style.css [hunk 4]
```diff
@@ -5,64 +5,138 @@ html {
...
-  margin: 2rem 0;
+  text-align: center;
 }
 
-ul {
-  flex-direction: column;
+.places_links {
+  width: 80%;
+  height: 10%;
+  margin-left: 10%;
```
Comment (m4v15) [inline public/style.css]: this breaks the arabic CSS when the text direction changes - use `margin:auto`

File: public/style.css [hunk 5]
```diff
@@ -5,64 +5,138 @@ html {
...
+  margin: 2% 0;
+  font-size: 1.5rem;
 }
 
 .place-wrap__links {
-  flex-direction: column;
+  width: 50%;
+  height: 20%;
+  justify-content: center;
+  margin: 0 0 2% 25%;
```
Comment (m4v15) [inline public/style.css]: see above ☝️

File: views/place-form.hbs [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+  <label>{{localText.placeHours}}
+    <input type="text" name="openingHours" required>
+  </label>
+
+  <label>{{localText.formOwnerId}}
+    <input type="text" name="ownerId">
+  </label>
+
+  <label>{{localText.formImageUrl}}
```
Comment (m4v15) [inline views/place-form.hbs:23]: I think maybe this shouldn't be required now

Comment (m4v15) [reply views/place-form.hbs:23]: Use case of someone adding a place without having an image url to hand

File: views/place-form.hbs [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    <input type="text" name="imageUrl" required>
+  </label>
+
+  <label>{{localText.placeWebsite}}
+    <input type="text" name="website" required>
```
Comment (m4v15) [inline views/place-form.hbs]: same as imageUrl

File: views/place-form.hbs [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+  </label>
+
+  <label>{{localText.placePhone}}
+    <input type="number" name="phone" required="">
```
Comment (m4v15) [inline views/place-form.hbs]: why `required=""`

File: views/place-form.hbs [hunk 9]
```diff
@@ -0,0 +1,128 @@
+<form class="add-form" action="/{{lang}}/add-place" method="post">
+  <h1>Add Place</h1>
```
Comment (m4v15) [inline views/place-form.hbs]: this text should be in `localText`

**Review by m4v15 — CHANGES_REQUESTED** [review]

**Comment by philawsophizing** [comment]
ref #12 #13 @m4v15 cheers :maple_leaf:

**Comment by philawsophizing** [comment]
ref #28 carried away, but happier with layout now. Smaller pr's going frwd. :maple_leaf:

File: views/places.hbs [hunk 10]
```diff
@@ -1,12 +1,12 @@
 
 <ul>
-  <h1 class="page-title">{{localText.placesTitle}}</h1>
+  <h1 class="page-title">{{localLang.placesTitle}}</h1>
   <div class="places_links">
-    <a href="../{{lang}}/add-place">{{localText.placesAdd}}</a><a href="../{{lang}}/edit-place">{{localText.placesEdit}}</a>
-
+    <a href="/{{../lang}}/add-place">Add Place</a><a href="{{../lang}}/edit-place">Edit Place</a>
```
Comment (m4v15) [inline views/places.hbs]: is this right? not using localLang?

**Review by m4v15 — CHANGES_REQUESTED** [review]

**Comment by m4v15** [comment]
@philawsophizing if you make those changes I'm happy for you to merge (as I won't be able to)

File: views/places.hbs (hunk 10 above)
Comment (philawsophizing) [reply views/places.hbs]: Yeah it was changed to localText in the middleware. Seemed less confusing than having locallLang and lang together.

**Pull request merged**